xdsl/transforms/experimental/ConvertStencilToLLMLIR.py

```diff
@@ -153,6 +155,30 @@ def verify_load_bounds(cast: CastOp, load: LoadOp):
         )
 
 
+class IndexOpToLoopSSA(RewritePattern):
+    @staticmethod
+    def discover_enclosing_loops(op: Operation) -> Sequence[Operation]:
+        loop_list: Sequence[Operation] = []
+        parent_op = op.parent_op()
+        if parent_op is not None:
+            loop_list.extend(IndexOpToLoopSSA.discover_enclosing_loops(parent_op))
```

Comment:
I'm sure that this won't be a problem in practice, because we don't expect hundreds of nested loops, but this is subtlely O(n²) due to creating n lists. One way to avoid this is to migrate this to an iterator function, changing the return type to `Iterable[Operation]`, changing this line to
```suggestion
            yield from IndexOpToLoopSSA.discover_enclosing_loops(parent_op)
```
and changing the line below to `yield op`.

The caller can then construct a single list from the iterable

Reply:
Nice! I've changed it as suggested :)